internal/provider/resource_instance.go

```diff
@@ -1183,6 +1251,115 @@ func deleteNICs(ctx context.Context, client *oxide.Client, models []instanceReso
 	return nil
 }
 
+// attachExternalIPs attaches the external IPs specified by externalIPs to the
+// instance specified by instanceID.
+func attachExternalIPs(ctx context.Context, client *oxide.Client, externalIPs []instanceResourceExternalIPModel, instanceID string) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	for _, v := range externalIPs {
+		externalIPID := v.ID
+		externalIPType := v.Type
+
+		switch oxide.ExternalIpKind(externalIPType.ValueString()) {
+		case oxide.ExternalIpKindEphemeral:
+			params := oxide.InstanceEphemeralIpAttachParams{
+				Instance: oxide.NameOrId(instanceID),
+				Body: &oxide.EphemeralIpCreate{
+					Pool: oxide.NameOrId(externalIPID.ValueString()),
+				},
+			}
+
+			if _, err := client.InstanceEphemeralIpAttach(ctx, params); err != nil {
+				diags.AddError(
+					fmt.Sprintf("Error attaching ephemeral external IP with ID %s", externalIPID.ValueString()),
+					"API error: "+err.Error(),
+				)
+
+				return diags
+			}
+
+		case oxide.ExternalIpKindFloating:
+			params := oxide.FloatingIpAttachParams{
+				FloatingIp: oxide.NameOrId(externalIPID.ValueString()),
+				Body: &oxide.FloatingIpAttach{
+					Kind:   oxide.FloatingIpParentKindInstance,
+					Parent: oxide.NameOrId(instanceID),
+				},
+			}
+
+			if _, err := client.FloatingIpAttach(ctx, params); err != nil {
+				diags.AddError(
+					fmt.Sprintf("Error attaching floating external IP with ID %s", externalIPID.ValueString()),
+					"API error: "+err.Error(),
+				)
+
+				return diags
+			}
+		default:
+			diags.AddError(
+				"Error attaching external IP",
+				fmt.Sprintf("Unknown external IP type %s", externalIPType.ValueString()),
```

Comment:
I know it's unlikely anyone will hit this error, but the message is a little cryptic. Can we have something that's a little more descriptive?

Reply:
Updated! I was going to remove this block entirely but I just updated the error text. Let me know if it reads better.